src/components/stepper/Stepper.tsx

```diff
@@ -0,0 +1,67 @@
+import StepperCircle from "./StepperCircle"
+import { MainContainer } from "./Stepper.styles";
+import { createRef, useRef, useState } from "react";
+import { Button, ButtonsContainer, StepperContainer, FormContainer } from "./StepperHandler.styles";
+
+type Step = {
+    title: string;
+    component: React.ForwardRefExoticComponent<any>;
+    props?: any;
+}
+
+
+interface StepperProps {
+    steps: Step[];
+}
+
+
+function Stepper({ steps }: StepperProps) {
+
+    const [currentStep, setCurrentStep] = useState<number>(0);
+    const formRefs = useRef(steps.map(() => createRef<any>()));
+
+    const nextStep = async () => {
+        const currentFormRef2 = formRefs.current[currentStep];
+        currentFormRef2.current.submit().then((res: any) => {
+            if (res) {
+                setCurrentStep((prev) => Math.min(prev + 1, steps.length - 1));
+            }
+        });
+    };
```

Comment:
Sobre esto, podemos ponerle directamente `currentFormRef` y cambiar la forma en la que estas seteando los pasos

```suggestion
    const nextStep = async () => {
        const currentFormRef = formRefs.current[currentStep];
        currentFormRef.current.submit().then((res: any) => {
            if (res) {
                setCurrentStep(prev => prev + 1);
            }
        });
    };
```